src/main/java/edu/harvard/iq/dataverse/util/json/JsonPrinter.java

```diff
@@ -580,6 +580,14 @@ public static JsonObjectBuilder json(MetadataBlock block, List<DatasetField> fie
         return blockBld;
     }
 
+    public static JsonArrayBuilder json(List<MetadataBlock> metadataBlocks, boolean returnDatasetFieldTypes, boolean onlyDisplayedOnCreate) {
+        JsonArrayBuilder arrayBuilder = Json.createArrayBuilder();
+        for (MetadataBlock metadataBlock : metadataBlocks) {
+            arrayBuilder.add(returnDatasetFieldTypes ? json(metadataBlock, onlyDisplayedOnCreate) : brief.json(metadataBlock));
```

Comment:
If returnDatasetFieldTypes is false and onlyDisplayedOnCreate is true, this will still return all metadatablocks, not just those displayed on create? Is that a bug or feature?

Reply:
I see that in use only displayed on create blocks are handed to this method when onlyDisplayedOnCreate is true - perhaps a comment here that that's a requirement?

Reply:
Actually the `onlyDisplayedOnCreate` param in this method was referring to the dataset field types that we want to print, not the metadata blocks, that's why I just renamed the parameter to `printOnlyDisplayedOnCreateDatasetFieldTypes`. Tell me if you think this makes the behavior clearer, if not I can add a comment as you suggest.

In practice, the method could be invoked for any metadata block, which is why I personally prefer not to add the comment which would relate its use to a specific scenario.

Reply:
That's works